### PR TITLE
fix(generator): update component scaffolding

### LINKED ISSUE #5775: Add component template to facilitate contribution
Add an empty/near empty basic component that is aimed to be a start point for converting a contributing component to PF's structure. 

Should contain:
- test folder, basic test file
- examples folder, basic md file
- component file (perhaps with basic props to show descriptions) (perhaps two variants, one function component and one class component)
- index.ts
- readme of some variety, explaining how to fill the above files and what else needs to be changed for PF to pick up the component (namely the main index.ts)

### PR DESCRIPTION
**Pull request by redallen**
**What**: Closes #5775

**Additional issues**:

### CODE REVIEW

File: scripts/generators/patternfly-4-component/templates/component/component.tsx.hbs [hunk 1]
```diff
@@ -7,10 +7,10 @@ interface {{ componentName }}Props {
   className?: string;
```
Comment (kmcfaul) [inline scripts/generators/patternfly-4-component/templates/component/component.tsx.hbs:7]: Add some basic prop descriptions? Would be helpful to show their structure + our docs look for them.

Comment (redallen) [reply scripts/generators/patternfly-4-component/templates/component/component.tsx.hbs:9]: Added!

**Review by dlabrecq — DISMISSED** [review]

File: scripts/generators/patternfly-4-component/templates/component/component.tsx.hbs [hunk 2]
```diff
@@ -3,17 +3,19 @@ import styles from '@patternfly/react-styles/css/{{typeDir}}/{{componentName}}';
...
-export const {{componentName}}: React.SFC<{{componentName}}Props> = ({
+export const {{componentName}}: React.FunctionComponent<{{componentName}}Props> = ({
   className,
   children
-}) => {
+}: {{ componentName }}Props) => {
   return (
-    <div {...props} className={css(styles.{{ camelCase componentName }}, props.className)}>
-      {props.children}
+    <div {...props} className={css(styles.{{ camelCase componentName }}, className)}>
```
Comment (tlabaj) [inline scripts/generators/patternfly-4-component/templates/component/component.tsx.hbs:17]: Can you move the props to be spread at the end.

**Review by tlabaj — CHANGES_REQUESTED** [review]
looks good so far.  Can you also add the `displayName`

File: scripts/generators/patternfly-4-component/templates/component/component.tsx.hbs (hunk 2 above)
Comment (redallen) [reply scripts/generators/patternfly-4-component/templates/component/component.tsx.hbs]: Done!

**Review by tlabaj — APPROVED** [review]

**Review by nicolethoen — APPROVED** [review]

**Pull request merged**